services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -167,45 +176,75 @@ spec = do
       context "no zuser" $ do
         it "responds with 'forbidden' and a helpful message" $ do
           env <- ask
-          callIdpCreate' ((env ^. teSpar)) Nothing (env ^. teNewIdp)
-            `shouldRespondWith` check (>= 400) [aesonQQ|{"error":"no auth token"}|]
+          callIdpCreate' (env ^. teSpar) Nothing (env ^. teNewIdp)
+            `shouldRespondWith` check (== 403) [aesonQQ|{"error":"no auth token"}|]
 
       context "zuser has no team" $ do
         it "responds with 'forbidden' and a helpful message" $ do
           env <- ask
-          (uid, _) <- call $ createRandomPhoneUser ((env ^. teBrig))
-          callIdpCreate' ((env ^. teSpar)) (Just uid) (env ^. teNewIdp)
-            `shouldRespondWith` check (>= 400) [aesonQQ|{"error":"you need to be team admin to create an IdP"}|]
+          (uid, _) <- call $ createRandomPhoneUser (env ^. teBrig)
+          callIdpCreate' (env ^. teSpar) (Just uid) (env ^. teNewIdp)
+            `shouldRespondWith` check (== 403) [aesonQQ|{"error":"you need to be team admin to create an IdP"}|]
 
-      context "zuser is a team member, not a team admin" $ do
+      context "zuser is a team member, but not a team admin" $ do
         it "responds with 'forbidden' and a helpful message" $ do
-          pending
+          env <- ask
+          (_owner, tid) <- call $ createUserWithTeam (env ^. teBrig) (env ^. teGalley)
+          nobody <- let Just perms = Galley.newPermissions mempty mempty
+                    in call $ createTeamMember (env ^. teBrig) (env ^. teGalley) tid perms
+          callIdpCreate' (env ^. teSpar) (Just nobody) (env ^. teNewIdp)
+            `shouldRespondWith` check (== 403) [aesonQQ|{"error":"you need to be team admin to create an IdP"}|]
```

Comment:
Similar to the previous comment, tests here could match on `label`s of errors, not on the exact string representation. Instead of duplicating the error strings both in the src and in the tests, defining errors in one place and re-using them in the tests would be preferable.